From 7f5b91685b001aaf73e490f0b6f47ee7d1855cd9 Mon Sep 17 00:00:00 2001
From: Brian Lin <brianhlin@gmail.com>
Date: Tue, 15 Nov 2016 10:38:42 -0600
Subject: [PATCH 1/2] Suppress locally run payload job records (SOFTWARE-2532)

If users set 'SuppressGridLocalRecords=1' in their ProbeConfig and use
the following, recommended HTCondor configuration, Gratia will ignore
jobs that complete on a non-glidein slot:

JOBGLIDEIN_ResourceName=\
"$$([IfThenElse(IsUndefined(TARGET.GLIDEIN_ResourceName), \
    IfThenElse(IsUndefined(TARGET.GLIDEIN_Site), \
        \"Local Job\",
        TARGET.GLIDEIN_Site),
    TARGET.GLIDEIN_ResourceName)])"
SUBMIT_EXPRS = $(SUBMIT_EXPRS) JOBGLIDEIN_ResourceName
---
 condor/condor_meter | 2 ++
 1 file changed, 2 insertions(+)

diff --git a/condor/condor_meter b/condor/condor_meter
index 4b2ce84..27144ca 100755
--- a/condor/condor_meter
+++ b/condor/condor_meter
@@ -742,6 +742,8 @@ def classadToJUR(classad):
             r.Grid("Campus", "Campus Factory Usage")
         elif campus_flock_usage.search(classad['MATCH_EXP_JOBGLIDEIN_ResourceName']):
             r.Grid("Campus", "Campus Flocking Usage")
+        elif classad['MATCH_EXP_JOBGLIDEIN_ResourceName'] == "Local Job":
+            r.Grid("Local", "Local execution based on ResourceName")
 
     if 'JobUniverse' in classad:
         # scheduler and local universes are always considered to be local
-- 
2.5.5

From 56145d313c218e3dbb7c66405a8506ae7fd0f9e0 Mon Sep 17 00:00:00 2001
From: Brian Lin <brianhlin@gmail.com>
Date: Tue, 15 Nov 2016 15:57:32 -0600
Subject: [PATCH 2/2] Correctly mark locally run payload jobs (SOFTWARE-2532)

---
 condor/99_gratia-gwms.conf | 5 +++--
 1 file changed, 3 insertions(+), 2 deletions(-)

diff --git a/condor/99_gratia-gwms.conf b/condor/99_gratia-gwms.conf
index d7fd43d..3fb9afa 100644
--- a/condor/99_gratia-gwms.conf
+++ b/condor/99_gratia-gwms.conf
@@ -9,8 +9,9 @@
 # a priority list of attributes in the startd's classad:
 # 1 - GLIDEIN_ResourceName - Using in the OSG's GlideinWMS Factories
 # 2 - GLIDEIN_Site - Used for all GlideinWMS instances to uniquely identify a site
-# 3 - FileSystemDomain - Should be defined on all Startd's, everywhere.
-JOBGLIDEIN_ResourceName="$$([IfThenElse(IsUndefined(TARGET.GLIDEIN_ResourceName), IfThenElse(IsUndefined(TARGET.GLIDEIN_Site), IfThenElse(IsUndefined(TARGET.FileSystemDomain), \"Local Job\", TARGET.FileSystemDomain), TARGET.GLIDEIN_Site), TARGET.GLIDEIN_ResourceName)])"
+# If neither are defined, it means that the job ran locally and these records can
+# be suppressed by setting 'SuppressGridLocalRecords="1"' in the ProbeConfig.
+JOBGLIDEIN_ResourceName="$$([IfThenElse(IsUndefined(TARGET.GLIDEIN_ResourceName), IfThenElse(IsUndefined(TARGET.GLIDEIN_Site), \"Local Job\", TARGET.GLIDEIN_Site), TARGET.GLIDEIN_ResourceName)])"
 
 # Add the JOBGLIDEIN_ResourceName to every job that is submitted from this host
 SUBMIT_EXPRS = $(SUBMIT_EXPRS) JOBGLIDEIN_ResourceName   
-- 
2.5.5

